src/main/java/io/opentracing/contrib/global/concurrent/TracedCallable.java

```diff
@@ -0,0 +1,124 @@
+import java.util.concurrent.Callable;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Convenience {@link Callable} wrapper that will execute within a new {@link Span} if an
+ * {@link #withOperationName(String) operationName} is provided.<br>
+ * If no operationName is provided, the call will be executed without starting a new Span.
+ *
+ * @author Sjoerd Talsma
+ */
+public class TracedCallable<T> implements Callable<T> {
+    private static final Logger LOGGER = Logger.getLogger(TracedCallable.class.getName());
+
+    private final Callable<T> delegate;
+    private String operationName = "";
+    private SpanContext parentContext = null;
+
+    protected TracedCallable(Callable<T> delegate) {
+        if (delegate == null) throw new NullPointerException("Callable delegate is <null>.");
+        this.delegate = delegate;
+    }
+
+    public static <T> TracedCallable<T> of(Callable<T> delegate) {
+        return new TracedCallable<T>(delegate);
+    }
+
+    public TracedCallable<T> withOperationName(String operationName) {
+        this.operationName = operationName;
+        return this;
+    }
+
+    public TracedCallable<T> asChildOf(SpanContext parent) {
+        this.parentContext = parent;
+        return this;
+    }
+
+    protected Span startNewSpan() {
+        SpanBuilder spanBuilder = GlobalTracer.tracer().buildSpan(operationName);
+        if (parentContext != null) spanBuilder = spanBuilder.asChildOf(parentContext);
+        return spanBuilder.start();
+    }
+
+    /**
+     * This method {@linkplain #startNewSpan() starts a new span} and calls the delegate, returning the result.
+     * Finally, the started span is {@linkplain Span#close() closed} again.
+     *
+     * @return The result from the delegate call.
+     * @throws Exception in case the delegate call threw an exception
+     *                   or there were exceptions starting or closing a new span.
+     */
+    public T call() throws Exception {
+        // The javadoc doesn't mention the details of the errorhandling
+        // because they are in line with normal try-with-resources logic:
+        // - try to close resources created (started span) whether the call succeeded or failed.
+        // - in case closing fails: a) throw the call exception if there was one (but addSuppressed the close failure)
+        //                       or b) if there was no call exception throw the close failure instead.
+
+        Span newSpan = null;            // so it can be closed
+        Exception callException = null; // so it can be thrown from finally if necessary
+        try {
+
+            newSpan = startNewSpan();
+            return delegate.call();
+
+        } catch (Exception callEx) {
+            callException = callEx;
+            throw callException;
+        } finally {
+            if (newSpan != null) try {
+                newSpan.close();
+            } catch (Exception closeFailure) {
```

Comment:
Should `span.close` throw exception or not. I think that should decide by the tracer implementation. As a globalTracer, we should only control the tracer-instance, spans, and in-process(in-thread/inter-thread) propagation. 
Let's leave it to **TRACER**.